TESTING.md

```diff
@@ -197,6 +198,84 @@ When the test runs unattended, it is possible to stop it gracefully by creating
 `test-artifacts` directory. When such file exists, the test execution stops at the next `terraform` command, waiting
 for user input.
 
+## Custom terraform scripts
+
+The commands `make test-binary-prepare` and `make test-binary` have the side effect of compiling custom Terraform scripts located in `./vcd/test-templates`.
+These tests are similar to the ones produced by the testing framework, but unlike the standard ones, they can be edited by users. And users vcan also remove and add files to suit their purposes.
+
+The files in `test-templates` are not executable directly by `terraform`: they need to be processed (which happens during `make binary-prepare`) and their placeholders expanded to the values taken from the configuration file.
+
+A **placeholder** is a label enclosed in double braces and prefixed by a dot, such as `{{.LabelName}}`. The template processor will replace that label with the corresponding value taken from the test configuration file.
+
+This is the list of placeholders that you can use in template files:
```

Comment:
I'd probably note that these variables are coming from `vcd_test_config` to remove some kind of magic feeling when the user does not know how our testing works and jumps straight to this section.

Reply:
I've improved the field description